### PR TITLE
get questions/answer for my game

### PR DESCRIPTION
**Pull request by FernandoArevaloE**
* start game functionality improved to get questions/answers from experience..
* column questions added in experience table
* column questions_detail in table game renamed to questions

when a game is started the response looks like:
```
{
    "game_id": "300a3803-c205-49cf-b9c6-a7337dd2fe49",
    "player_id": "6c6377c3-f755-4cbf-8e73-cae093a4e65b",
    "experience_id": "1504b003-0dea-4804-a106-f039e80ff1d7",
    "questions": [
        {
            "id": 3,
            "question": "question 3?",
            "answers": [
                {
                    "id": 1,
                    "text": "aaaa"
                },
                {
                    "id": 2,
                    "text": "aaaa"
                },
                {
                    "id": 2,
                    "text": "aaaa"
                }
            ]
        },
        {
            "id": 29,
            "question": "question 29?",
            "answers": [
                {
                    "id": 1,
                    "text": "aaaa"
                },
                {
                    "id": 2,
                    "text": "aaaa"
                },
                {
                    "id": 2,
                    "text": "aaaa"
                }
            ]
        },
        {
            "id": 15,
            "question": "question 15?",
            "answers": [
                {
                    "id": 1,
                    "text": "aaaa"
                },
                {
                    "id": 2,
                    "text": "aaaa"
                },
                {
                    "id": 2,
                    "text": "aaaa"
                }
            ]
        },
        {
            "id": 24,
            "question": "question 24?",
            "answers": [
                {
                    "id": 1,
                    "text": "aaaa"
                },
                {
                    "id": 2,
                    "text": "aaaa"
                },
                {
                    "id": 2,
                    "text": "aaaa"
                }
            ]
        },
        {
            "id": 28,
            "question": "question 28?",
            "answers": [
                {
                    "id": 1,
                    "text": "aaaa"
                },
                {
                    "id": 2,
                    "text": "aaaa"
                },
                {
                    "id": 2,
                    "text": "aaaa"
                }
            ]
        }
    ],
    "score": "0",
    "started": "2024-05-11T02:33:38.728Z"
}
```

### CODE REVIEW

File: app/init.sql [hunk 1]
```diff
@@ -54,6 +54,7 @@ CREATE TABLE experience (
     active BOOLEAN DEFAULT TRUE,
     actor_id uuid NOT NULL, 
     name VARCHAR NOT NULL,
+    questions jsonb DEFAULT '{"data":[]}'::jsonb NOT NULL,
```
Comment (andymotagl) [inline app/init.sql:57]: I guess this can be just the []

File: app/src/service/experience.js [hunk 2]
```diff
@@ -43,9 +43,38 @@ const getCurrentExperience = async (playerId) => {
     return experience;
 }
 
+const getQuestionsForGame = async (experienceId, questions, exclude) => {
+    const experience = await getExperienceById(experienceId);
+
+    if (!questions || questions <= 0) {
```
Comment (andymotagl) [inline app/src/service/experience.js:49]: Should not this be an attribute of experience?

File: app/src/service/experience.js [hunk 3]
```diff
@@ -43,9 +43,38 @@ const getCurrentExperience = async (playerId) => {
...
+        throw new BadRequest('Invalid number of questions');
+    }
+
+    const availableQuestions = exclude ? experience.questions.data.filter(q => !exclude.find(q.id)) : experience.questions.data;
```
Comment (andymotagl) [inline app/src/service/experience.js:53]: Let us put this in 3 different lines 
- The condition
- True return with a new line on each dot 
- False retun

File: app/src/service/experience.js [hunk 4]
```diff
@@ -43,9 +43,38 @@ const getCurrentExperience = async (playerId) => {
...
+
+    const randomQuestions = [];
+
+    for (var i = 0; i < questions && availableQuestions.length > 0; i++) {
```
Comment (andymotagl) [inline app/src/service/experience.js:57]: Condition questions can be removed because it was evaluated before

**Comment by andymotagl** [comment]
My concern with this request is I would not know if the frontend would be able to render the questions/answers in the same shot when starting the game.

File: app/src/service/experience.js (hunk 2 above)
Comment (FernandoArevaloE) [reply app/src/service/experience.js:49]: yes, it is. questions is the amount of questions to be selected

File: app/src/service/experience.js (hunk 3 above)
Comment (FernandoArevaloE) [reply app/src/service/experience.js:53]: done

File: app/src/service/experience.js (hunk 2 above)
Comment (andymotagl) [reply app/src/service/experience.js:49]: What I mean is that the experience is collected but the value taking here is the one passed as parameter.

File: app/init.sql (hunk 1 above)
Comment (FernandoArevaloE) [reply app/init.sql]: done

File: app/src/service/experience.js (hunk 2 above)
Comment (FernandoArevaloE) [reply app/src/service/experience.js]: done

File: app/src/service/experience.js (hunk 4 above)
Comment (FernandoArevaloE) [reply app/src/service/experience.js]: done

**Review by andymotagl — APPROVED** [review]

**Pull request merged**